projects/plugins/boost/app/modules/optimizations/lcp/class-lcp.php

```diff
@@ -257,6 +318,10 @@ private function optimize_viewport( $buffer, $lcp_data ) {
 			return $buffer;
 		}
 
+		if ( $lcp_data['type'] === 'unknown' ) {
+			return $buffer;
+		}
+
```

Comment:
Instead of this, it might be beneficial to check the type against a known list of types that work. This would maintain compatibility in case a new type is added on the cloud but the plugin version is lagging behind. For example, we add support for a new type in the cloud, let's call it, "youtube-embed". Then we release a plugin update to support that type. However, sites with older version of the plugin will continue to work with the result if we take the approach below.

```suggestion
		if ( in_array( $lcp_data['type'], array( 'background-image', 'img' ), 1 ) {
			return $buffer;
		}

```

Reply:
Good point, I've updated it now to return early if it's not a type the plugin knows how to handle.